community/modules/scripts/htcondor-install/files/autoscaler.py

```diff
@@ -260,45 +258,59 @@ def scale(self):
         currentTarget = int(responseGroupInfo["targetSize"])
         print("Current MIG target size: " + str(currentTarget))
 
+        currentActions= responseGroupInfo["currentActions"]
+        n_being_born = currentActions["creating"] + \
+            currentActions["recreating"] + \
+            currentActions["starting"] + \
+            currentActions["restarting"] + \
+            currentActions["verifying"]
+
+        # count nodes that are either idle and have joined the pool or are
+        # still joining the pool from the MIG; there is some potential for
+        # overlap with the "verifying" state that will be addressed by
+        # only deleting nodes whose uptime exceeds this period of overlap
+        n_idle = len(idle_node_ads)
+        n_idle_total = n_being_born + n_idle
+        print(f"Idle nodes found: {n_being_born} birthing and {n_idle} idling")
+
+        # always keep size tending toward the minimum idle VMs requested
+        delta = min_hosts_for_queue - currentTarget
+        if delta < self.compute_instance_min_idle:
+            min_hosts_for_queue += self.compute_instance_min_idle - delta - n_idle_total
+            print(f"Adjusting target to maintain idle VMs: {min_hosts_for_queue}")
+
+        if min_hosts_for_queue > self.compute_instance_limit:
+            self.size = self.compute_instance_limit
+            print(f"MIG target size will be limited by {self.compute_instance_limit}")
+        else:
+            self.size = min_hosts_for_queue
+
+        print("New MIG target size: " + str(self.size))
```

Comment:
nit. Rest of this code uses f-strings, let's keep it consistent, just to drop `str` cast.

Reply:
This has been captured in a backlog task for modernizing the style and addressing linting issues.